tokio/src/runtime/task/list.rs

```diff
@@ -111,25 +111,36 @@ impl<S: 'static> OwnedTasks<S> {
             // to the field.
             task.header().set_owner_id(self.id);
         }
+        self.push_inner(id, task, notified)
+    }
 
-        let mut lock = self.inner.lock();
-        if lock.closed {
+    #[inline]
+    pub(crate) fn push_inner(
+        &self,
+        task_id: super::Id,
+        task: Task<S>,
+        notified: Notified<S>,
+    ) -> Option<Notified<S>>
+    where
+        S: Schedule,
+    {
+        let mut lock = self.lists[task_id.0 as usize & (self.grain - 1) as usize].lock();
```

Comment:
this code for indexing the set of task lists by a task ID is somewhat complex, and it's repeated in a few places. what do you think about potentially adding a function that takes a task ID and returns a `MutexGuard` so that we don't have to repeat this?

Reply:
Yes, this is duplicate code, I have move the way of get list by task id into the method `segment_inner`.